sonar-python-plugin/src/test/java/org/sonar/plugins/python/indexer/SonarQubePythonIndexerTest.java

```diff
@@ -326,6 +326,17 @@ public void test_regular_scan_when_scan_without_parsing_fails() {
     assertThat(logTester.logs(LoggerLevel.INFO)).contains("1/1 source file has been analyzed");
   }
 
+  @Test
+  public void test_no_data_in_cache_for_parse_error() {
+    file1 = createInputFile(baseDir, "parse_error.py", InputFile.Status.ADDED, InputFile.Type.MAIN);
+
+    List<InputFile> inputFiles = new ArrayList<>(List.of(file1));
+
+    pythonIndexer = new SonarQubePythonIndexer(inputFiles, cacheContext);
+    pythonIndexer.buildOnce(context);
+    assertThat(writeCache.getData().get(projectSymbolTableCacheKey("moduleKey:parse_error.py"))).isNull();
```

Comment:
Am I correct that the issue was that we did put `null` into the cache? I think it would make sense to have an assertion that checks that the cache does not contain `null` for this file.